src/project/internal/projectactionscontroller.cpp

```diff
@@ -854,6 +854,21 @@ bool ProjectActionsController::saveProjectLocally(const io::path_t& filePath, Sa
         return false;
     }
 
+    // Prevents Save operation from overwriting read-only files
+    QFileInfo savePath(project->path().toQString());
+    if (savePath.exists() && !savePath.isWritable() && saveMode == SaveMode::Save) {
+        warnScoreCouldnotBeSaved(trc("project/save", "This file is read-only."));
+        return false;
+    }
+
+    else if (saveMode == SaveMode::SaveAs || saveMode == SaveMode::SaveCopy) {
+        QFileInfo saveAsPath(filePath.toQString());
+        if (saveAsPath.exists() && !saveAsPath.isWritable()) {
+            warnScoreCouldnotBeSaved(trc("project/save", "This file is read-only."));
+            return false;
+        }
+    }
+
```

Comment:
Why not
```cpp
    // Prevents Save operation from overwriting read-only files
    QFileInfo savePath(project->path().toQString());
    if (saveMode == SaveMode::SaveAs || saveMode == SaveMode::SaveCopy) {
        saveAsPath(() =filePath.toQString();
    }
    if (savePath.exists() && !savePath.isWritable() && saveMode == SaveMode::Save) {
        warnScoreCouldnotBeSaved(trc("project/save", "This file is read-only."));
        return false;
    }

```

Reply:
Maybe even exclude this for Windows and Mac

Reply:
Thank for your input! I agree with your suggestion. I'm assuming before the else if you meant } and not ], right?

Reply:
Also why not doing the entire check inside `save()`?

Reply:
Yes, of course  }

Reply:
Or even better do it in `doSave`, there os a check for writability already:

```cpp
    // Step 1: check writable
    {
#if defined(Q_OS_WIN) || defined(Q_OS_MAC) 
        if (fileSystem()->exists(savePath) && !fileSystem()->isWritable(savePath)) {
#else
        QFileInfo savePath(path.toQString());
        if (savePath.exists() && !savePath.isWritable()) {
#endif
            LOGE() << "failed save, not writable path: " << savePath;
            return make_ret(notation::Err::UnknownError);
        }
```

Reply:
Actually it might be as easy as
```cpp
    // Step 1: check writable
    {
        if (fileSystem()->exists(savePath) && !fileSystem()->isWritable(savePath)
              && fileSystem()->exists(targetContainerPath) && !fileSystem()->isWritable(targetContainerPath)) {
            LOGE() << "failed save, not writable path: " << savePath;
            return make_ret(notation::Err::UnknownError);
        }
```

Reply:
> Actually there (in `doSave()`) it might be as easy as
> 
> ```c++
>     // Step 1: check writable
>     {
>         if ((fileSystem()->exists(savePath) && !fileSystem()->isWritable(savePath)) ||
>             (fileSystem()->exists(targetContainerPath) && !fileSystem()->isWritable(targetContainerPath))) {
>             LOGE() << "failed save, not writable path: " << targetContainerPath;
>             return make_ret(notation::Err::UnknownError);
>         }
> ```
> 
> a 1-line addition...

Thanks. This solution is indeed cleaner. Tested on Linux and it works, will test on Windows now. However, I feel like the user should have more information when this error occurs, instead of "unknown error". What do you think of creating an Err for this verification with a text like "This file is read-only" or "This file or folder is not writable".

Reply:
Even shorter:
´´´cpp
    // Step 1: check writable
    {
        if (!fileSystem()->isWritable(savePath) || !fileSystem()->isWritable(targetContainerPath))) {
            LOGE() << "failed save, not writable path: " << targetContainerPath;
            return make_ret(notation::Err::UnknownError);
        }
```

Reply:
For a proper error message:
```cpp
#include "global/io/ioretcodes.h"
    // Step 1: check writable
    {
        if (fileSystem()->exists(savePath) && !fileSystem()->isWritable(savePath)) {
            LOGE() << "failed save, not writable path: " << savePath;
            return make_ret(mu::io::Err::FSWriteError);
        }

        if (ioMode == engraving::MscIoMode::Dir) {
            // Dir needs to be created, otherwise we can't move to it
            if (!QDir(targetContainerPath).mkpath(".")) {
                LOGE() << "Couldn't create container directory";
                return make_ret(mu::io::Err::FSMakingError);
            }
        }
    }
```

Reply:
These may not be the final error codes, but are the closest existing ones I could find. Maybe ioretcodes.h needs to get extended with some better onesbut these do seem to be io errors rather than notation errors